lib/routes/recovery-keys.js

```diff
@@ -0,0 +1,230 @@
+        const recoveryKeyId = request.params.recoveryKeyId
+        let recoveryData
+
+        customs.check(request, 'getRecoveryKey')
+          .then(getRecoveryKey)
+          .then(() => reply({recoveryData}), reply)
+
+        function getRecoveryKey() {
+          if (sessionToken.tokenVerificationId) {
+            throw errors.unverifiedSession()
+          }
+
+          return db.getRecoveryKey(uid, recoveryKeyId)
+            .then((res) => recoveryData = res.recoveryData)
+        }
+      }
+    },
+    {
+      method: 'POST',
+      path: '/account/reset/recoveryKeys',
+      config: {
+        auth: {
+          strategy: 'accountResetToken'
+        },
+        validate: {
+          query: {
+            keys: isA.boolean().optional()
+          },
+          payload: {
+            authPW: isA.string().min(64).max(64).regex(HEX_STRING).required(),
+            wrapKb: isA.string().min(64).max(64).regex(HEX_STRING).required(),
+            recoveryKeyId: isA.string().max(256).required(),
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Password.changeRecoveryKeys', request)
+        const authPW = request.payload.authPW
+        const wrapKb = request.payload.wrapKb
+        const recoveryKeyId = request.payload.recoveryKeyId
+        const uid = request.auth.credentials.uid
+
+        let authSalt, password, verifyHash, sessionToken, keyFetchToken, account
+
+        checkRecoveryKey()
+          .then(getAccount)
+          .then(passwordReset)
+          .then(notifyAccount)
+          .then(createSessionToken)
+          .then(createKeyFetchToken)
+          .then(deleteRecoveryKey)
+          .then(createResponse)
+          .then(reply, reply)
+
+        function checkRecoveryKey() {
+          return db.getRecoveryKey(uid, recoveryKeyId)
+        }
+
+        function getAccount() {
+          return db.account(uid)
+            .then((result) => account = result)
+        }
+
+        // Password reset using a recovery key behaves similarly to a password change. The client should have
+        // retrieved their original `kB` and rewrapped.
+        function passwordReset() {
+          return random.hex(32)
+            .then(hex => {
+              authSalt = hex
+              password = new Password(authPW, authSalt, verifierVersion)
+              return password.verifyHash()
+            })
+            .then((hash) => {
+              verifyHash = hash
+              return password.wrap(wrapKb)
+            })
+            .then((wrapWrapKb) => {
+              return db.resetAccount(
+                {uid},
+                {
+                  verifyHash: verifyHash,
+                  authSalt: authSalt,
+                  wrapWrapKb: wrapWrapKb,
+                  verifierVersion: password.version
+                }
+              )
+            })
+        }
+
+        function notifyAccount() {
+          return Promise.resolve()
+        }
```

Comment:
Is this a placeholder for some future code that is to come, or is it a remnant of some old code that came before?